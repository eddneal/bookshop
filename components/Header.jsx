import React from 'react';
import { Menu } from 'semantic-ui-react';
import PropTypes from 'prop-types';

import { Link } from '../routes/routes';

export const navItems = [
  {
    name: 'Home',
    route: 'index',
  },
  {
    name: 'About',
    route: 'about',
  },
];

const Header = props => (
  <header>
    <nav>
      <Menu pointing secondary>
        {navItems.map(item => (
          <Menu.Item active={props.page === item.name} key={item.name}>
            <Link route={item.route}>
              <a>{item.name}</a>
            </Link>
          </Menu.Item>
        ))}
      </Menu>
    </nav>
  </header>
);

Header.propTypes = {
  page: PropTypes.string.isRequired,
};

export default Header;
