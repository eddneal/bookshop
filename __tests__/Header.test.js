import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Header, { navItems } from '../components/Header';

describe('With Enzyme', () => {
  it('Header outputs navItems as links', () => {
    const app = shallow(<Header page="Home" />);
    navItems.forEach((item, navIndex) => {
      app.find('a').forEach((elem, elemIndex) => {
        if (navIndex === elemIndex) {
          expect(elem.text()).toEqual(item.name);
        }
      });
    });
  });
});

describe('With Snapshot Testing', () => {
  it('Header matches Snapshot', () => {
    const component = renderer.create(<Header page="About" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
