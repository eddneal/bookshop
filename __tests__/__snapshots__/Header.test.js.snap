// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Header matches Snapshot 1`] = `
<header>
  <nav>
    <div
      className="ui pointing secondary menu"
    >
      <div
        className="item"
        onClick={[Function]}
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
      <div
        className="active item"
        onClick={[Function]}
      >
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </div>
    </div>
  </nav>
</header>
`;
